x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/snapshotlifecycle/action/GetSnapshotLifecycleAction.java

```diff
@@ -146,22 +152,34 @@ public boolean equals(Object obj) {
      * {@link SnapshotLifecyclePolicyMetadata}, however, it elides the headers to ensure that they
      * are not leaked to the user since they may contain sensitive information.
      */
-    public static class SnapshotLifecyclePolicyItem implements ToXContentFragment {
+    public static class SnapshotLifecyclePolicyItem implements ToXContentFragment, Writeable {
 
         private final SnapshotLifecyclePolicy policy;
         private final long version;
         private final long modifiedDate;
-
-        public SnapshotLifecyclePolicyItem(SnapshotLifecyclePolicy policy, long version, long modifiedDate) {
-            this.policy = policy;
-            this.version = version;
-            this.modifiedDate = modifiedDate;
+        @Nullable
+        private final Long lastSuccessDate;
+        @Nullable
+        private final Long lastFailureDate;
+        @Nullable
+        private final String lastFailureInfo;
+
+        public SnapshotLifecyclePolicyItem(SnapshotLifecyclePolicyMetadata policyMetadata) {
+            this.policy = policyMetadata.getPolicy();
+            this.version = policyMetadata.getVersion();
+            this.modifiedDate = policyMetadata.getModifiedDate();
+            this.lastSuccessDate = policyMetadata.getLastSuccessDate();
+            this.lastFailureDate = policyMetadata.getLastFailureDate();
+            this.lastFailureInfo = policyMetadata.getLastFailureInfo();
```

Comment:
I'm also thinking, we *may* want to separate this into two different objects so they are clearly defined and separated, something like:


```
{
    ... policy json etc ...
    "last_success": {
        "snapshot_name": "production-snapshot-2019-03-15-kwno8wplqfaswa5rsbvdvg",
        "time": 1553813657,
        "time_string": "2019-03-15.21:09:00-blah"
    },
    "last_failure": {
        "snapshot_name": "production-snapshot-2018-12-02-aoeu890gaoeuhaoeu098as",
        "time": 1553629109,
        "time_string": "2018-12-02.19:23:01",
        "failure": "java.lang.NotCoolEnoughException: you are not cool enough to take snapshots\n\torg.try.to.take.the.snapshot.DoIt:129\n\t...etc..."
    }
}
```

I'm trying to think of the easiest thing to consume, I think having only dates and an error (which only pertains to the failure, rather than the success, will be a little confusing.

What do you think about output like this?

Reply:
I had considered doing that originally, and only didn't since there were only a couple fields. If we're going to include more information (which I agree would be good, I didn't think of the snapshot name) then yes, I'll switch these to object fields instead.